src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -962,38 +1232,99 @@ function AnalyticsService:GetRemoteConfig(remote: RemoteConfig): { [any]: any }
 		elseif not remote.defaultValue then
 			return reject("AnalyticsService.GetRemoteConfig - remote.defaultValue is required")
 		end
-		
+
 		if not self._enabled then
 			return resolve(remote.defaultValue)
 		end
-		
+
 		local player: Player? = remote.player
-		local context: any = self:_getServerPsuedoPlayer()
-		local server: any = not player and HttpApi:initRequest(
-			self._options.gameKey,
-			self._options.secretKey,
-			self._options.build,
-			context.PlayerData,
-			""
-		)
-		
+		local context: ServerPsuedoPlayer = self:_getServerPsuedoPlayer()
+		local server = if player == nil
+			then -- Using Luau conditional expression so type can be inferred https://luau-lang.org/syntax#if-then-else-expressions
+				GameAnalytics.HttpApi:initRequest(
+					self._options.gameKey,
+					self._options.secretKey,
+					self._options.build,
+					context.PlayerData,
+					""
+				)
+			else nil
+
 		if server and server.statusCode >= 9 then
 			for _, config in (server.body.configs or {}) do
 				if config.key == remote.name then
 					return resolve(config.value)
 				end
 			end
 		end
-		
+
 		if player and not GameAnalytics:isRemoteConfigsReady(player.UserId) then
 			return resolve(remote.defaultValue)
 		end
-		
-		return resolve(player and GameAnalytics:getRemoteConfigsValueAsString(player.UserId, {
+
+		resolve(player and GameAnalytics:getRemoteConfigsValueAsString(player.UserId, {
 			key = remote.name,
-			defaultValue = remote.defaultValue
+			defaultValue = remote.defaultValue,
 		}) or remote.defaultValue)
 	end)
 end
 
-return AnalyticsService
\ No newline at end of file
+-- GameAnalytics method aliases for custom dimensions
+local dimensionSetter: { [string]: string } = {
+	dimension01 = "setCustomDimension01",
+	dimension02 = "setCustomDimension02",
+	dimension03 = "setCustomDimension03",
+}
+
+--[=[
+	Used to set a custom dimension for a player
+
+	Example Use:
+	```lua
+	AnalyticsService:SetCustomDimension({
+		userId = player.UserId,
+		dimension = "dimension01",
+		value = "value"
+	})
+	```
+
+	To remove a custom dimension from a player, set the value to "".
+
+	For more information about dimensions, refer to [GameAnalytics docs](https://docs.gameanalytics.com/integrations/sdk/roblox/sdk-features?_highlight=dimension#custom-dimensions) on dimensions.
+
+	@param data DimensionData
+	@return Promise
```

Comment:
```suggestion
	@return Promise<T>
```
Unless the type is already defined, you need to define custom return types with Moonwave by denoting them with a `<T>` after the type (Eg. `Promise<T>`)